src/client/dfuse/dfuse_core.c

```diff
@@ -76,6 +87,17 @@ ir_free(struct d_hash_table *htable, d_list_t *rlink)
 
 /* Inode entry hash table operations */
 
+static uint32_t
+ih_rec_hash(struct d_hash_table *htable, d_list_t *link)
+{
+	const struct dfuse_inode_entry	*ie;
+
+	ie = container_of(link, struct dfuse_inode_entry, ie_htl);
+
+	return d_hash_string_u32((const char *)&ie->ie_stat.st_ino,
```

Comment:
I'm picking on dfuse here because it's the one I know but why does this use a string function here to get the hash of a 64 bit integer?

Specifically for this case we assign the value of st_ino and it's set locally on the node starting at zero so simply using the lower bits would be ideal, in fact I need to look at bucket distribution in dfuse for JIRA-4928

Reply:
The hash function for key is not specified and used default hash generation function which is d_hash_string_u32().

Modified it according your suggestion.